src/Orleans/Async/BatchWorker.cs

```diff
@@ -132,5 +132,53 @@ public bool IsIdle()
             else if (waitfortask != null)
                 await waitfortask;
         }
+
+        /// <summary>
+        /// Notify the worker that there is more work, and wait for the current work cycle, and also the next work cycle if there is currently unserviced work.
+        /// </summary>
+        public async Task NotifyAndWaitForWorkToBeServiced()
+        {
+            Task<Task> waitForTaskTask = null;
+            Task waitForTask = null;
+
+            lock (this)
```

Comment:
yeah, I don't usually do this. But here, the lock is actually exposed on purpose because code that is enqueueing work uses it.  

Still, I don't like it much myself. It's probably a better idea for code maintenance to use a protected member, so only subclasses get access to the lock. I will do that.